controllers/machinedeployment_controller.go

```diff
@@ -111,7 +111,7 @@ func (r *MachineDeploymentReconciler) reconcile(ctx context.Context, d *clusterv
 		if d.Status.ObservedGeneration < d.Generation {
 			patch := client.MergeFrom(d.DeepCopy())
 			d.Status.ObservedGeneration = d.Generation
-			if err := r.Status().Patch(ctx, d, patch); err != nil {
+			if err := r.Client.Status().Patch(ctx, d.DeepCopy(), patch); err != nil {
```

Comment:
Could you please add a comment indicating why we're operating on a copy?

Reply:
Also, would it be better to avoid doing the DeepCopy twice (it's also on line 112)?

Reply:
Will add a comment, but we need to deep copy twice, otherwise the if we try to use the same deep copy as the patch the patch will result in a noop.

Reply:
Can't we use the patch helper here and in the other places instead of issuing individual patches?

Reply:
> Can't we use the patch helper here and in the other places instead of issuing individual patches?

Yes, but without massive refactoring it will be doing much more work than is needed since each individual place we are issuing patches or updates now do not update both status and non-status.